lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/UmAppDatabase.kt

```diff
@@ -1259,12 +1264,83 @@ abstract class UmAppDatabase : DoorDatabase(), SyncableDoorDatabase {
             }
         }
 
+        /**
+         * Add fields required for class and school codes for students to join a class or school
+         *  Changes from 37-38:
+        1. Added personGroupUid to Person
+        2. Added personGroupFlag to PersonGroup
+        3. Removed groupPersonUid from PersonGroup
+         */
+        val MIGRATION_38_39 = object : DoorMigration(38, 39) {
+            override fun migrate(database: DoorSqlDatabase) {
+
+                if (database.dbType() == DoorDbType.SQLITE) {
+                    //Person
+                    database.execSQL("ALTER TABLE Person ADD COLUMN personGroupUid BIGINT DEFAULT 0 NOT NULL")
+
+                    //PersonGroup
+                    database.execSQL("ALTER TABLE PersonGroup RENAME to PersonGroup_OLD")
+                    database.execSQL("CREATE TABLE IF NOT EXISTS PersonGroup (  " +
+                            "groupMasterCsn  BIGINT NOT NULL , groupLocalCsn  BIGINT NOT NULL, " +
+                            "groupLastChangedBy  INTEGER NOT NULL, groupName  TEXT , " +
+                            "groupActive  INTEGER NOT NULL , personGroupFlag  INTEGER NOT NULL, " +
+                            "groupUid  INTEGER NOT NULL PRIMARY KEY  AUTOINCREMENT  NOT NULL )")
+                    database.execSQL("INSERT INTO PersonGroup (" +
+                            "groupUid, groupMasterCsn, groupLocalCsn, groupLastChangedBy, " +
+                            "groupName, groupActive) SELECT groupUid, groupMasterCsn, " +
+                            "groupLocalCsn, groupLastChangedBy, groupName, groupActive " +
+                            "FROM PersonGroup_OLD")
+                    database.execSQL("DROP TABLE PersonGroup_OLD")
+
+
+                } else if (database.dbType() == DoorDbType.POSTGRES){
+                    //Person
+                    database.execSQL("ALTER TABLE Person RENAME to Person_OLD")
```

Comment:
I would recommend using Postgres' improved support for table modification here instead.

Reply:
Also: for any table (e.g. PersonGroup) where on Sqlite you had to rename/drop tables, you need to add the SQLite triggers back (otherwise it won't sync again)

e.g.
```
_stmt.executeUpdate("""
          |CREATE TRIGGER UPD_43
          |AFTER UPDATE ON PersonGroup FOR EACH ROW WHEN
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(NEW.groupMasterCsn = 0 
          |OR OLD.groupMasterCsn = NEW.groupMasterCsn
          |)
          |ELSE
          |(NEW.groupLocalCsn = 0  
          |OR OLD.groupLocalCsn = NEW.groupLocalCsn
          |) END)
          |BEGIN 
          |UPDATE PersonGroup SET groupLocalCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN NEW.groupLocalCsn 
          |ELSE (SELECT MAX(MAX(groupLocalCsn), OLD.groupLocalCsn) + 1 FROM PersonGroup) END),
          |groupMasterCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(SELECT MAX(MAX(groupMasterCsn), OLD.groupMasterCsn) + 1 FROM PersonGroup)
          |ELSE NEW.groupMasterCsn END)
          |WHERE groupUid = NEW.groupUid
          |; END
          """.trimMargin())
          _stmt.executeUpdate("""
          |CREATE TRIGGER INS_43
          |AFTER INSERT ON PersonGroup FOR EACH ROW WHEN
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(NEW.groupMasterCsn = 0 
          |
          |)
          |ELSE
          |(NEW.groupLocalCsn = 0  
          |
          |) END)
          |BEGIN 
          |UPDATE PersonGroup SET groupLocalCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN NEW.groupLocalCsn 
          |ELSE (SELECT MAX(groupLocalCsn) + 1 FROM PersonGroup) END),
          |groupMasterCsn = 
          |(SELECT CASE WHEN (SELECT master FROM SyncNode) THEN 
          |(SELECT MAX(groupMasterCsn) + 1 FROM PersonGroup)
          |ELSE NEW.groupMasterCsn END)
          |WHERE groupUid = NEW.groupUid
          |; END
          """.trimMargin())
```